drivers/esp8266/sock_udp/udp.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (C) 2016 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the UDP socket with the esp8266 802.11b/g/n radio module
+ *              In the implementation we assume that we are only woking with exactly one ESP8266 module.
+ *              The implementation follows the monitor pattern, thereby allow only one udp action at a time.
+ *
+ * @author      Markus Blechschmidt <markus.blechschmidt@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "net/sock/udp.h"
+#include "esp8266.h"
+#include "xtimer.h"
+#include "thread.h"
+#include "mutex.h"
+#include "sema.h"
+#include "msg.h"
+#include "net/ipv4/addr.h"
+#include "net/af.h"
+
+#include "gnrc_sock_internal.h"
+
+#define ENABLE_DEBUG  (0)
+#include "debug.h"
+
+#define INTERVAL_MS (1)
+
+#define TEMP_ID_STR "temp"
+#define LOCAL_ID_STR "local"
+#define REMOTE_ID_STR "remote"
+
+/* private varibaled fo the driver */
+static ESP8266_t esp8266;
+static char threadstack[THREAD_STACKSIZE_DEFAULT + DEBUG_EXTRA_STACKSIZE];
+static mutex_t esp_sock_mutex;
+static mutex_t esp_callback_mutex;
+// static kernel_pid_t api_pid;
+static void *send_data;
+static size_t send_len;
+static void *rece_data;
+static size_t rece_len;
+static sema_t sem_loc_rece_count;
+static sema_t sem_loc_rece_space;
+static ESP8266_Connection_t *temp_conn;
+static ESP8266_Connection_t *local_conn;
+static ESP8266_Connection_t *remote_conn;
+static sock_udp_ep_t current_ep;
+
+#ifdef MODULE_GNRC_SOCK_CHECK_REUSE
+static sock_udp_t *_udp_socks = NULL;
```

Comment:
Speaking of: where is your `sock_types.h` where `sock_udp_t` type for `esp8266_sock_udp` is supposed to be defined?